components/social-media-icon/index.js

```diff
@@ -0,0 +1,31 @@
+import { IconMapper } from './social-media.constant';
+import PropTypes from 'prop-types';
+import React from 'react';
+import classNames from './social-media-icon.module.scss';
+
+const SocialMediaIcon = (props) => {
+  const { id, type } = props;
+
+  const onClick = (e) => {
+    e.stopPropagation();
+  };
+
+  return (
+    <a
+      onClick={onClick}
+      className={classNames.iconContainer}
+      target="_blank"
+      rel="noreferrer"
+      tabIndex="0"
+      href={`${IconMapper[type].href}/${[id]}`}>
+      <img className={classNames.image} alt={IconMapper[type].alt} src={IconMapper[type].src} />
+    </a>
+  );
+};
+
+SocialMediaIcon.propTypes = {
```

Comment:
if props are not required, default it using `defaultProps`

Reply:
yup, i would like to keep the proptypes for better understanding of the props

Reply:
I was asking for something like this (if these are not required props). So they don't break your code.

SocialMediaIcon.propTypes = {
  id: PropTypes.string,
  type: PropTypes.string
};

SocialMediaIcon.defaultProps = {
  id: '',
  type: ''
};

Reply:
done